lib/chef/provider/package/chocolatey.rb

```diff
@@ -0,0 +1,190 @@
+#
+# Copyright:: Copyright (c) 2015 Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require 'chef/provider/package'
+require 'chef/resource/chocolatey_package'
+require 'chef/mixin/powershell_out'
+
+class Chef
+  class Provider
+    class Package
+      class Chocolatey < Chef::Provider::Package
+        include Chef::Mixin::PowershellOut
+
+        provides :chocolatey_package, os: "windows"
+
+        # Declare that our arguments should be arrays
+        package_class_supports_arrays
+
+        # Responsible for building the current_resource and as a (necessary)
+        # side effect this populates the candidate_version Array.
+        #
+        # @return [Chef::Resource::ChocolateyPackage] the current_resource
+        def load_current_resource
```

Comment:
load_current_resource isn't going to run until converge, in Chef::Provider#run_action, right? Can we check if chocolately is installed here and raise if it isn't?

Is it "necessary" to populate candidate_version here because we're going to run #define_resource_requirements next where we make sure there are candidates for for "all" packages?